File: Backend/Events/Converters/Journals/QuestConverter.cs
using Backend.Domain.Models.Journals;
using Backend.Events.DTO.Journal;

namespace Backend.Events.Converters.Journals;

public static class QuestConverter
{
    public static QuestDTO ToDTO(Quest quest) => new()
    {
        Id = quest.Id,
        Requisites = quest.Requisites.Select(RequisiteConverter.ToDTO).ToList(),
        Steps = quest.Steps.Select(QuestStepConverter.ToDTO).ToList()
    };
}

File: Frontend/src/converters/QuestConverter.ts
import type { QuestDTO } from '../events/dto/Journals/QuestDTO';
import type { Quest } from '../models/Journal';
import { RequisiteConverter } from './RequisiteConverter';
import { StepConverter } from './StepConverter';

export class QuestConverter {
    public static convert(dto: QuestDTO | null): Quest | null {
        if (!dto || !dto.id) return null;
        return {
            id: dto.id,
            title: '', // Preenchido via getLocalizedQuest no composable de localização
            description: '',
            prerequisites: dto.requisites ? dto.requisites.map(r => RequisiteConverter.convert(r)) : [],
            steps: dto.steps ? dto.steps.map(s => StepConverter.convert(s)) : []
        };
    }
}
